### ISSUE TITLE
On 32bit OS, files greater than 2GB will not tar up correctly, leaving unusable tar ball.

### ISSUE THREAD

**Issue by JuanitoMoore** [tklauser/libtar#3]
I got this to work by substituting struct stat with struct stat64 and lstat with lstat64. This can possibly be done with -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE CFLAGS.

Minor: The libtar -g -z -t <file> will show a negative number, so change the output format from %9ld to %9lu: printf("%9lu ", th_get_size(t));

I could send git diff with my changes, if you'd like. I also added flags O_LARGEFILE for open, although that might not be needed.

**Comment by tklauser** [comment]
Yes, a patch for this would certainly be welcome as long as it won't break any of the supported OSes. Do you know whether all of the OSes listed in the README support struct stat64 and lstat64?

**Comment by xloem** [comment]
I've also noticed that `int_to_oct` and `oct_to_int` use "int" and "long" as their types which could unnecessarily limit range on 32-bit systems.  I would use "unsigned long long" for these values.

I'll try to open an untested pull request so people of future years can work off this, too.

**Comment by xloem** [comment]
> Yes, a patch for this would certainly be welcome as long as it won't break any of the supported OSes. Do you know whether all of the OSes listed in the README support struct stat64 and lstat64?

It looks like the CFLAGS enable operating system dependent features to treat `stat` as `stat64`, so there would simply be no effect on operating systems without stat64.
